src/controllers/krate/delete.rs

```diff
@@ -147,6 +166,33 @@ async fn has_rev_dep(conn: &mut AsyncPgConnection, crate_id: i32) -> QueryResult
     Ok(rev_dep.is_some())
 }
 
+/// Email template for notifying a crate owner about a crate being deleted.
+///
+/// The owner usually should be aware of the deletion since they initiated it,
+/// but this email can be helpful in detecting malicious account activity.
+#[derive(Debug, Clone)]
+struct CrateDeletionEmail<'a> {
+    user: &'a str,
+    krate: &'a str,
+}
+
+impl Email for CrateDeletionEmail<'_> {
+    fn subject(&self) -> String {
+        format!("crates.io: Deleted \"{}\" crate", self.krate)
+    }
+
+    fn body(&self) -> String {
+        format!(
+            "Hi {},
+
+this is a confirmation email for the deletion of your \"{}\" crate.
```

Comment:
Wording tweak (feel free to ignore it if you want, but in that case let's at least capitalise the first letter):

```suggestion
Your \"{}\" crate has been deleted, per your request.
```